src/session/session_api.c

```diff
@@ -530,6 +534,8 @@ __session_create_readonly(
 	WT_UNUSED(uri);
 	WT_UNUSED(config);
 
+	WT_STAT_FAST_CONN_INCR((WT_SESSION_IMPL *)wt_session,
```

Comment:
@sulabhM Casting a ``WT_SESSION`` to a ``WT_SESSION_IMPL`` uses a certain protocol in the source code at the moment. It always happens via a local variable even if it's only used in one place.

While looking at this code, I realised that there is another problem with the read-only stubs, which is that they don't use the ``SESSION_API_CALL`` macros - which mean that errors will be reported incorrectly. The changes in the readonly stubs would be better if they had the form:

```
--- a/src/session/session_api.c
+++ b/src/session/session_api.c
@@ -531,12 +531,18 @@ static int
 __session_create_readonly(
     WT_SESSION *wt_session, const char *uri, const char *config)
 {
+       WT_SESSION_IMPL *session;
+
        WT_UNUSED(uri);
        WT_UNUSED(config);

-       WT_STAT_FAST_CONN_INCR((WT_SESSION_IMPL *)wt_session,
-           session_table_create_fail);
-       return (__wt_session_notsup(wt_session));
+       session = (WT_SESSION_IMPL *)wt_session;
+       SESSION_API_CALL_NOCONF(session, create);
+
+
+       WT_STAT_FAST_CONN_INCR(session, session_table_create_fail);
+       ret = __wt_session_notsup(wt_session);
+err:   API_END_RET(session, ret);
 }

 /*
```